### ISSUE TITLE
Suggest the use of Github Discussions

### ISSUE THREAD

**Issue by TJMusiitwa** [flutter/flutter#60696]
**More of a feature request for the Flutter GitHub Repository...**

With the announcement of Github Discussions, it is my hope that the Flutter repository is able to adopt this as a way of potentially reducing the number of issues that are likely to relate to a question that can be easily answered by community  and as a way of as Github puts it in their blog [post](https://github.blog/2020-05-06-new-from-satellite-2020-github-codespaces-github-discussions-securing-code-in-private-repositories-and-more/#discussions), 

> "... _a community’s knowledge base grows naturally. And because discussions aren’t closed the way issues are, they can easily serve as a place for maintaining FAQs_ ..."

Of course when this feature comes out of Beta or sooner...
Many thanks to the Flutter team and the community

**Comment by Sunbreak** [comment]
**Could we use Discussion?**

GitHub Discussion is in beta and open up to every repo. Below is one of my example repo:

<img width="342" alt="Screen Shot 2020-12-09 at 3 12 33 PM" src="https://user-images.githubusercontent.com/7928961/101596971-fd73d480-3a30-11eb-81e5-a7143c243d49.png">

TestFlight of iOS App is available as well

https://twitter.com/_ryannystrom/status/1336422008062685189

![image](https://user-images.githubusercontent.com/7928961/101597279-770bc280-3a31-11eb-835f-06174087bc12.png)

**Comment by Hixie** [comment]
In general we already have borderline too many places where communications take place (we just had a survey about this very topic in fact). Between GitHub, our Discord server, e-mail, and other channels like the community Discord server, Reddit, Twitter, StackOverflow, corporate chat systems, personal e-mail, etc, it's already chaos with nobody really knowing where to go when they want to communicate with the team.

As a result, I suspect we will not be enabling this feature in the near future.

**Comment by YazeedAlKhalaf** [comment]
I believe that enabling this feature will make it easier as every thing in one platform would be great.
This feature enables:
- Ability to ask questions and get answers from experts, flutter team and also the community
- Ability to make a discussion about any topic you would like to talk about, such as a place where people can share their work, another discussion for getting one-to-one help, etc...

This combines both Stack Overflow and Discord into one place where people won't be lost.

Before pushing this proposal to the future we could create a testing repo and do some testing and see if it fits. 🚀

This issue has been here sitting for a very long time with no progress, But if discussions where there we could have a category for proposals for triaging before creating an issue. It will be way better and I always come to the issues tab to find someone who has a personal problem in code and opens an issue for it and it gets closed. That is not useful to neither the issue author or the members with contributer access.

**Comment by iapicca** [comment]
sorry for the [duplicate](https://github.com/flutter/flutter/issues/72013), posting here the **_use case_** and **_proposal_** from my issue

> ## Use case
> 
> - many new issues are not bug report feature request, but **_generic questions_** or request for **_help to implement a feature_**
> 
> - I [honestly believed](https://github.com/flutter/flutter/issues/61094) that the option `I want help writing my application` pointing to [stackoverflow.com](https://stackoverflow.com/) 
> would reduce the problem, but I clearly I was wrong, 
> this might depend from different reasons:
>    - maybe some reporter does not want to use a platform different than **github**
>    - maybe some reporter is not satisfied with the experience on [stackoverflow.com](https://stackoverflow.com/) 
> 
> ## Proposal
> 
> github has recently introduced [discussions](https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/about-team-discussions), I do not have a direct experience with it, but it [looks nice](https://github.com/vercel/vercel/discussions) 
> (I'm not affiliated with that repo, just the first result from google)
> 
> please consider investigate this solution to help reducing invalid issues
> 
> 
>

**Comment by YazeedAlKhalaf** [comment]
Sorry for the duplicate, my issue proposal 😉

> I suggest trying the discussion new feature in GitHub so people can have conversation without opening issues. It might look > > like Stack Overflow but in the same repo but I think we should give it a try. Because then engineers could take a look and > > > answer some questions there or discuss other stuff.

> It will make personal issues disappear, because people usually look for experts in the field to help them that is why they come > to here and open an issue but if we could enable discussion in the repo which is a new feature in GitHub it will really lift some > load off the issues tab 🚀

> If you have never seen discussions UI, here is an example in my repo:
> link for live try: [https://github.com/YazeedAlKhalaf/Flutter_Installer/discussions](https://github.com/YazeedAlKhalaf/Flutter_Installer/discussions)
> ![image](https://user-images.githubusercontent.com/41989385/101596323-eec3f800-3a05-11eb-9b70-82f6f2ccc233.png)
> ![image](https://user-images.githubusercontent.com/41989385/101596393-069b7c00-3a06-11eb-81da-41528f59d73f.png)

**Comment by Hixie** [comment]
> * Ability to ask questions and get answers from experts, flutter team and also the community

This is already possible on the community Discord, our own Discord, reddit, StackOverflow, the flutter-dev mailing list, and (if all else fails) filing issues. It's not clear to me why adding yet another mechanism would improve matters?


> * Ability to make a discussion about any topic you would like to talk about, such as a place where people can share their work, another discussion for getting one-to-one help, etc...

That's already possible in all the places I list above and many, many more. Why would adding one more such place improve matters?

**Comment by TJMusiitwa** [comment]
Ideally each of the above different mediums serves a different purpose and/or audience that is for sure. 
The flutter dev mailing list could be the medium for updates on the framework.
I see StackOverflow as the medium where all the questions are asked but GitHub Discussions could act as the overall "textbook" where the most voted/helpful answers are placed, to act as a sort of _"First place to look"_ before looking elsewhere.
But also speaking from my own perspective and not that of others, I personally don't have a Discord and Reddit account and don't want to use/join Discord/Reddit. 
So possibly the use of Discussions would be a lot more favourable to me but then again...this should be up for discussion with the community at large to see how it can be guided.

**Comment by robertpiosik** [comment]
I'd like to rise my hand in this issue.

What is missing in Discord is **searchability** in Google.

I strongly believe that GitHub Discussions is an amazing place for ideas to evolve. The great example of such adoption is [Next.js discussions page](https://github.com/vercel/next.js/discussions).

**Comment by Hixie** [comment]
There are two groups of people for whom we would consider using Discussions:
* The developer community (people using Flutter).
* The team (people creating Flutter).

The developer community is hundreds of thousands of people and we do not have the bandwidth to moderate a discussion forum for a community that size. Instead, we let other groups self-form on Reddit, StackOverflow, Twitter, and countless other places. (We do have a mailing list though it's not clear to me if we should keep it to be honest, it might be worth shutting that down.) If people are interested in using GitHub Discussions for this, we would encourage the creation of different GitHub organizations to host those. We're happy to point to them from our list of communities once they are established with a good tone and a critical mass of users: https://flutter.dev/community

For the team, we use the issues database, Discord, Google Docs, and other existing tools and it's not clear that Discussions would be especially useful as an additional medium.

**Comment by robertpiosik** [comment]
I totally get your point but I don't think discussions require any moderation from the team. Author of a thread eventually marks it as answered and that's it.

Another positive aspect of enabling GitHub Discussions I'd like to bring is the fact that having greater amount of people willing to help others closer to a codebase make them many huge steps closer to help you, the team, with the framework by contributing to a codebase.

**Comment by Hixie** [comment]
Given the amount of moderation we have to do just on GitHub issues, I assure you we would have to do plenty of moderation in a discussion forum. :-)

**Comment by robertpiosik** [comment]
A discussion thread is what an issue is not :^)

Nothing bad with unanswered discussion thread. Take a minute to grasp a feel of discussions by this [example](https://github.com/vercel/next.js/discussions/11272). I love how subthreads of subquestions are formed without losing the ability to read through all answers. Also linking to issues is helpful and very easy as we are within the same platform to where mainline development are going.

**Comment by robertpiosik** [comment]
A discussion thread is what an issue is not :^)

Nothing bad with unanswered discussion thread. Take a minute to grasp a feel of discussions by this example. I love how subthreads of subquestions are formed without losing the ability to read through all answers. Also linking to issues is helpful and very convenient as we are within the same platform to where mainline development is going.

Voting for the best answer is a great feature also to say thank you to an answerer what might be encouraging to him/her. It's just a great mix of features of other platforms in the right place :^)

**Comment by Hixie** [comment]
Moderation isn't about making sure threads have answers, it's about applying the code of conduct.

**Comment by robertpiosik** [comment]
Yeah, applying code of conduct is a thing in a project of this size. Nevertheless bringing community closer to a codebase might bring many good things and we will never know if we don't try :)

**Comment by Hixie** [comment]
This is the internet, Reddit is a tab away from GitHub. I really don't think it would make that much difference if the URL they opened has "github.com" in the address bar instead of "stackoverflow.com" or some such. Anyone who wants to contribute already has all the information they need to do so, and many of us are active in those other fora as well (e.g. I participate on Reddit threads about Flutter quite often).

That said, if you can commit to providing the people to moderate a discussion forum with the potential to grow to hundreds of thousands of users, then I have no objection to exploring setting up the Discussions page. I just don't want to start down that path without a clear plan in place for how we will continue to maintain the welcoming tone of the conversation.
